Dev container used to create the Qt 5 AppImage does not build anymore

The `Dockerfiles/appimage-qt5` container now fails to build [while we're installing `aqtinstall`](https://github.com/nuttyartist/notes/blob/75c876ed96c7da64bbe98cd44cd590e91f364712/Dockerfiles/appimage-qt5#L43-L45):

```
#9 6.692   Building wheel for pyzstd (PEP 517): started
#9 6.869   Building wheel for pyzstd (PEP 517): finished with status 'error'
#9 6.869   ERROR: Command errored out with exit status 1:
#9 6.869    command: /usr/bin/python3 /tmp/tmpr0koulp2 build_wheel /tmp/tmp7l3uhxpc
#9 6.869        cwd: /tmp/pip-install-pzbi33_n/pyzstd
#9 6.869   Complete output (49 lines):
#9 6.869   running bdist_wheel
#9 6.869   running build
#9 6.869   running build_py
#9 6.869   creating build
#9 6.869   creating build/lib.linux-x86_64-cpython-38
#9 6.869   creating build/lib.linux-x86_64-cpython-38/pyzstd
#9 6.869   copying src/seekable_zstdfile.py -> build/lib.linux-x86_64-cpython-38/pyzstd
#9 6.869   copying src/zstdfile.py -> build/lib.linux-x86_64-cpython-38/pyzstd
#9 6.869   copying src/__init__.py -> build/lib.linux-x86_64-cpython-38/pyzstd
#9 6.869   copying src/__main__.py -> build/lib.linux-x86_64-cpython-38/pyzstd
#9 6.869   creating build/lib.linux-x86_64-cpython-38/pyzstd/c
#9 6.869   copying src/c/__init__.py -> build/lib.linux-x86_64-cpython-38/pyzstd/c
#9 6.869   copying src/__init__.pyi -> build/lib.linux-x86_64-cpython-38/pyzstd
#9 6.869   copying src/py.typed -> build/lib.linux-x86_64-cpython-38/pyzstd
#9 6.869   running build_ext
#9 6.869   +--------------------------------------------+
#9 6.869   |             Pyzstd build config            |
#9 6.869   +-------------------------+------------------+
#9 6.869   | Pyzstd version          | 0.16.2           |
#9 6.869   +-------------------------+------------------+
#9 6.869   | Implementation          | C                |
#9 6.869   +-------------------------+------------------+
#9 6.869   | Link to zstd library    | Statically link  |
#9 6.869   +-------------------------+------------------+
#9 6.869   | Enable AVX2/BMI2        | False            |
#9 6.869   +-------------------------+------------------+
#9 6.869   | Debug build             | False            |
#9 6.869   +-------------------------+------------------+
#9 6.869   | Warning as error        | False            |
#9 6.869   +-------------------------+------------------+
#9 6.869   building 'pyzstd.c._zstd' extension
#9 6.869   creating build/temp.linux-x86_64-cpython-38
#9 6.869   creating build/temp.linux-x86_64-cpython-38/src
#9 6.869   creating build/temp.linux-x86_64-cpython-38/src/bin_ext
#9 6.869   creating build/temp.linux-x86_64-cpython-38/zstd
#9 6.869   creating build/temp.linux-x86_64-cpython-38/zstd/lib
#9 6.869   creating build/temp.linux-x86_64-cpython-38/zstd/lib/common
#9 6.869   creating build/temp.linux-x86_64-cpython-38/zstd/lib/compress
#9 6.869   creating build/temp.linux-x86_64-cpython-38/zstd/lib/decompress
#9 6.869   creating build/temp.linux-x86_64-cpython-38/zstd/lib/dictBuilder
#9 6.869   x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -DPYZSTD_STATIC_LINK -DZSTD_MULTITHREAD -Izstd/lib -Izstd/lib/common -Izstd/lib/dictBuilder -I/usr/include/python3.8 -c src/bin_ext/pyzstd.c -o build/temp.linux-x86_64-cpython-38/src/bin_ext/pyzstd.o -g0 -flto
#9 6.869   In file included from src/bin_ext/pyzstd.c:1:
#9 6.869   src/bin_ext/pyzstd.h:8:10: fatal error: Python.h: No such file or directory
#9 6.869       8 | #include "Python.h"
#9 6.869         |          ^~~~~~~~~~
#9 6.869   compilation terminated.
#9 6.869   /tmp/pip-build-env-bf8r66tu/overlay/lib/python3.8/site-packages/setuptools/_distutils/dist.py:268: UserWarning: Unknown distribution option: 'test_suite'
#9 6.869     warnings.warn(msg)
#9 6.869   error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
#9 6.869   ----------------------------------------
#9 6.870   ERROR: Failed building wheel for pyzstd
#9 6.870 Successfully built patch
#9 6.870 Failed to build pyzstd
#9 6.940 ERROR: Could not build wheels for pyzstd which use PEP 517 and cannot be installed directly
```

This is a fairly recent failure, and it's caused by `pyzstd` (a transitive dependency of `aqtinstall`) deprecating Python 3.8 support (present on Ubuntu 20.04, used to build the Qt 5 AppImage) in [version 0.16.2](https://github.com/Rogdham/pyzstd/commit/9a6efb0cc406367c4ab686890d5fabe2677dc9cc), and therefore requiring its wheel to be built from source on the machine, instead of downloading a prebuilt wheel for Python 3.8, like it was before that version.

To fix this, installing `python3-dev` before trying to install `aqtinstall` should likely be enough.

--- a/Dockerfiles/appimage-qt5
+++ b/Dockerfiles/appimage-qt5
@@ -32,11 +32,11 @@ ARG DEBIAN_FRONTEND=noninteractive
 # - libxcb-xkb1: Used as dependency of the resulting AppImage.
 # - libxkbcommon-x11-0: Used as dependency of the resulting AppImage.
 # - make: Used to help build the application.
-# - python3/python3-pip: Used to install and run aqtinstall.
+# - python3/python3-dev/python3-pip: Used to install and run aqtinstall.
 RUN apt-get update && \
     apt-get install -y --no-install-recommends appstream cmake desktop-file-utils file git g++ libdbus-1-3 \
     libegl1 libfontconfig1 libglib2.0-0 libgl-dev libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-randr0 \
-    libxcb-render-util0 libxcb-shape0 libxcb-xinerama0 libxcb-xkb1 libxkbcommon-x11-0 make python3 python3-pip && \
+    libxcb-render-util0 libxcb-shape0 libxcb-xinerama0 libxcb-xkb1 libxkbcommon-x11-0 make python3 python3-dev python3-pip && \
     apt-get clean && \
     rm -rf /var/lib/apt/lists
 

